openpype/hosts/nuke/api/plugin.py

```diff
@@ -808,6 +817,18 @@ def __init__(self,
         self.log.info("File info was set...")
 
         self.file = self.fhead + self.name + ".{}".format(self.ext)
+        if ".{}".format(self.ext) not in VIDEO_EXTENSIONS:
+            # filename would be with frame hashes if
+            # the file extension is not in video format
+            filename = get_head_filename_without_hashes(
+                self.path_in, self.name)
+            self.file = filename
+            # make sure the filename are in
+            # correct image output format
+            if ".{}".format(self.ext) not in self.file:
```

Comment:
I suspect that we're always expecting the file should END with the extension - right? So we should still change the extension if the extension `.exr` for the filename was originally `hello.exr.0001.png`.

This should've been:
```python
if not self.file.endswith(".{}".format(ext)):
    # Swap the extension
    ...
```